core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt

```diff
@@ -896,7 +896,15 @@ object UnitActions {
         if (!tile.canPillageTile()) return false
         val tileOwner = tile.getOwner()
         // Can't pillage friendly tiles, just like you can't attack them - it's an 'act of war' thing
-        return tileOwner == null || unit.civInfo.isAtWarWith(tileOwner)
+        if (tileOwner != null) {
+            return unit.civInfo.isAtWarWith(tileOwner)
+        } else {
+            if (tile.canPillageTile())
+                return true
+            if (tile.canPillageRoad() && tile.roadOwner != "" && unit.civInfo.isAtWarWith(tile.getRoadOwner()!!))
+                return true
+            return false
+        }
```

Comment:
If we have a 'if return' we don't need the 'else' afterwards, this can be done with no nesting at all :)

Reply:
Done